File: Know Basic Maths/divisors.py
def print_divisors(num):
    div = []

    i = 1
    while i * i <= num:
        if num % i == 0:
            div.append(i)
            if i * i != num:
                div.append(num // i)
        i += 1

    for d in div:
        print(d, end=" ")
    print()


num = int(input("Enter a number\n"))
print_divisors(num)

File: Know Basic Maths/divisors.cpp
#include<bits/stdc++.h>
using namespace std;

void print_divisors(int num){
       vector<int>div;
         for(int i=1;i*i<=num;i++){
               if(num%i==0){
                     div.push_back(i);
                     if(i*i !=num){
                          div.push_back(num/i);
                     }
               }
         }
         for(auto it:div){
              cout<<it<<" ";
         }
         cout<<endl;
}

int main(){
       int num;
       cout<<"Enter a number "<<endl;
       cin>>num;
       print_divisors(num);
}